src/wp-admin/includes/class-wp-upgrader.php

```diff
@@ -527,7 +528,10 @@ public function install_package( $args = array() ) {
 			set_time_limit( 300 );
 		}
 
-		if ( empty( $source ) || empty( $destination ) ) {
+		if (
+			( ! is_string( $source ) || '' === trim( $source ) || trim( $source ) !== $source ) ||
```

Comment:
A couple of things:
- Ideally, we wouldn't run `trim()` twice per variable.
- We could `trim()` values with leading/traling spaces rather than error on these.

We could use these conditions:

```php
if ( '' === $source || '' === $destination ) {}
```

And instead change the assignments a few lines above to:
```php
$source      = is_string( $args['source'] ) ? trim( $args['source'] ) : '';
$destination = is_string( $args['destination'] ) ? trim( $args['destination'] ) : '';
```

This avoids multiple `trim()` calls, keeps the error conditions minimal, and cleans up the source and destination for use later in the method.

Alternatively, use the same assignments as above, but continue to error out for leading/trailing space:

```php
if (
    '' === $source || $source !== $args['source'] || 
    '' === $destination || $destination !== $args['destination']
) {}
```

Thoughts?

Reply:
It occurs to me that with the modified code `'' === $source` would suffice as the trimmed version would differ for whitespace only values.

I'm happy to follow your preference here as a component maintainer. Which would you prefer?

Reply:
We can either gracefully trim the values and continue, or error out if the trimmed doesn't match the original.

I'm definitely leaning towards erroring out for safety. I'd rather that if somehow this resulted in a BC break of an edge case, that the BC break was an early bail rather than due to continuing with an incorrect path.

Reply:
@costdev I've emoved the duplicate trim & retained the error in 945a1ba0c2d61497795cec90bfd1dc37e875d883